src/lib/Scenes/Artwork/Components/CommercialButtons/__tests__/BidButton-tests.tsx

```diff
@@ -0,0 +1,137 @@
+      const wrapper = await getWrapper(artwork)
+
+      expect(wrapper.text()).toContain("Registration pending")
+    })
+
+    it("with registered bidder", async () => {
+      const artwork = merge({}, ArtworkFromAuctionPreview, RegisteredBidder)
+      const wrapper = await getWrapper(artwork)
+
+      expect(wrapper.text()).toContain("Registration complete")
+    })
+  })
+
+  describe("for open auction", () => {
+    it("with open registration and not registered bidder ", async () => {
+      const artwork = merge({}, ArtworkFromTimedAuctionRegistrationOpen, NotRegisteredToBid)
+      const wrapper = await getWrapper(artwork)
+
+      expect(wrapper.text()).toContain("Bid")
+    })
+
+    it("with closed registration and not registered bidder ", async () => {
+      const artwork = merge({}, ArtworkFromTimedAuctionRegistrationClosed, NotRegisteredToBid)
+      const wrapper = await getWrapper(artwork)
+
+      expect(wrapper.text()).toContain("Registration closed")
+    })
+
+    it("with bidder registration pending approval", async () => {
+      const artwork = merge({}, ArtworkFromTimedAuctionRegistrationOpen, BidderPendingApproval)
+      const wrapper = await getWrapper(artwork)
+
+      expect(wrapper.text()).toContain("Registration pending")
+    })
+
+    it("with registered bidder", async () => {
+      const artwork = merge({}, ArtworkFromTimedAuctionRegistrationOpen, RegisteredBidder)
+      const wrapper = await getWrapper(artwork)
+
+      expect(wrapper.text()).toContain("Bid")
+    })
+
+    it("with registered bidder with bids", async () => {
+      const artwork = merge({}, ArtworkFromTimedAuctionRegistrationOpen, RegistedBidderWithBids)
+      const wrapper = await getWrapper(artwork)
+
+      expect(wrapper.text()).toContain("Increase max bid")
+    })
+  })
+
+  describe("for live auction", () => {
+    it("with open registration and not registered bidder ", async () => {
+      const artwork = merge({}, ArtworkFromLiveAuctionRegistrationOpen, NotRegisteredToBid)
+      const wrapper = await getWrapper(artwork)
+
+      expect(wrapper.find(Button).text()).toContain("Enter live bidding")
+    })
+
+    it("with closed registration and not registered bidder ", async () => {
+      const artwork = merge({}, ArtworkFromLiveAuctionRegistrationClosed, NotRegisteredToBid)
+      const wrapper = await getWrapper(artwork)
+
+      expect(wrapper.text()).toContain("Registration closed")
+      expect(wrapper.find(Button).text()).toContain("Watch live bidding")
+    })
+
+    it("with bidder registration pending approval", async () => {
+      const artwork = merge({}, ArtworkFromLiveAuctionRegistrationOpen, BidderPendingApproval)
+      const wrapper = await getWrapper(artwork)
+
+      expect(wrapper.find(Button).text()).toContain("Enter live bidding")
+    })
+
+    it("with registered bidder", async () => {
+      const artwork = merge({}, ArtworkFromLiveAuctionRegistrationOpen, RegisteredBidder)
```

Comment:
#nonblocking why use lodash's merge instead of `{..spreading}`?

Reply:
Going to keep as is for now.